Overview.bs

```diff
@@ -295,43 +295,44 @@ expansion of the template:
   <tr>
     <td>id</td>
     <td>
-      If the input id is numeric then this is the  numeric value converted to a string in hexadecimal representation
-      (using the digits 0-9, A-F). No padding with 0's is used. Otherwise, this is the string id value.
+      The input id encoded as a [[rfc4648#section-7|base32hex]] string (using the digits 0-9, A-V) with
+      padding omitted. When the id is an unsigned integer only the significant bytes are encoded. (For
+      example, when the integer is less than 256 only one byte is encoded.)
     </td>
   </tr>
   <tr>
     <td>d1</td>
     <td>
       The last character of the string in the id variable.
-      If id variable is empty then, the value is the character 0 (U+0030).
+      If id variable is empty then, the value is the character _ (U+005F).
     </td>
   </tr>
   <tr>
     <td>d2</td>
     <td>
       The second last character of the string in the id variable.
-      If the id variable has less than 2 characters then, the value is the character 0 (U+0030).
+      If the id variable has less than 2 characters then, the value is the character _ (U+005F).
     </td>
   </tr>
   <tr>
     <td>d3</td>
     <td>
       The third last character of the string in the id variable.
-      If the id variable has less than 3 characters then, the value is the character 0 (U+0030).
+      If the id variable has less than 3 characters then, the value is the character _ (U+005F).
     </td>
   </tr>
   <tr>
     <td>d4</td>
     <td>
       The fourth last character of the string in the id variable.
-      If the id variable has less than 4 characters then, the value is the character 0 (U+0030).
+      If the id variable has less than 4 characters then, the value is the character _ (U+005F).
     </td>
```

Comment:
Thinking about this a bit more, an alternative option to id_raw, would be to have a second variable (id64?) that uses a base64 url encoding with padding of the raw id string bytes. This is slightly less flexible for implementations, but should still allow a dynamic implementation to be built and base64url is pretty much optimal for encoding data into a url. If we go this route, we'll need to change how the id strings are interpreted. Instead of treating them as string of unicode code points, they would just be opaque binary data.

Reply:
This option would also reduce the encoded size of the mapping table (since we're storing the raw bytes vs the base64 encoded version), which is nice.